src/crypto/CHIPCryptoPAL.h

```diff
@@ -90,6 +93,16 @@ enum class CHIP_SPAKE2P_ROLE : uint8_t
     PROVER   = 1, // Commissioner
 };
 
+typedef struct P256PrivateKey
+{
+    uint8_t bytes[kP256_PrivateKey_Length];
+} P256PrivateKey;
+
+typedef struct P256PublicKey
```

Comment:
What is the scope of use of this new type? Is the idea that it should be used everywhere in CHIP code? For example, as an input to the ECDSA_validate_msg_signature() function. 

If we do want this type to be generally used across the entire CHIP codebase then there are few questions to answer:
1. What is the appropriate type for "bytes": array or a pointer? In some applications it is more convenient when it is a pointer so there is no need to move data around and just point to a place in message buffer or certificate.
2. should this structure be specific to P256 curve or general to any EC curve? 

Similar comment/question applies to P256PrivateKey type.

Reply:
I am targeting Session Establishment use case with this API (where the accessory generates a P256 key pair). 

The data type can be expanded for other uses. This will require changes to the APIs that have already been implemented and integrated, but it can be done.

1. The bytes array is ok, as we are passing references to the top level data structure. It helps with not moving data around. If we reuse this data type in other APIs, the same scheme must be followed to avoid unnecessary data copy.
2. Is there a CHIP usecase where we need other curves? If so, instead of having a static structure of fixed byte array, we can define an abstract class as the data type. Different curves can have implementation of that abstract type.

Reply:
1. SGTM
2. Today CHIP only supports P256 curve. I was thinking about future extensibility of the code to support other curves. Maybe it is something to deal with in the future...